ports/esp32/mphalport.h

```diff
@@ -55,7 +55,8 @@ extern TaskHandle_t mp_main_task_handle;
 extern ringbuf_t stdin_ringbuf;
 
 // Check the ESP-IDF error code and raise an OSError if it's not ESP_OK.
-void check_esp_err(esp_err_t code);
+#define check_esp_err(code) check_esp_err_(code, __FUNCTION__, __LINE__, __FILE__)
```

Comment:
This is the bit that's expensive for flash, it has to capture and store all the filename information.

This bit would need to be in the #if debug switch.

Then you run into discoverability issues... will future C developers find this to know to switch it on?

It's also somewhat inconsistent to add this kind of stuff just to esp port, what about C developers using other ports?

Reply:
I think the size concern is resolved by @IhorNehrutsa, at least I now don't see any increase building locally.

> It's also somewhat inconsistent to add this kind of stuff just to esp port, what about C developers using other ports?

I guess `mp_raise_OSError` is the most useful port-agnostic API where this could be added. It could be nice to have the same error reporting there, I certainly sometimes wonder which of a few possible places an OSError bubbled up from.

I personally think it's OK to add this extra functionality just for these ESP-IDF APIs now though, as these APIs have no equivalent on other ports. However I can see the argument both ways (it's confusing if some `OSError`s come with extra context and others do not.)